pkg/ring/ring.go

```diff
@@ -256,24 +256,28 @@ func (r *Ring) GetAll() (ReplicationSet, error) {
 		return ReplicationSet{}, ErrEmptyRing
 	}
 
-	ingesters := make([]IngesterDesc, 0, len(r.ringDesc.Ingesters))
-	maxErrors := r.cfg.ReplicationFactor / 2
+	// Calculate the number of required ingesters;
+	// ensure we always require at least RF-1 when RF=3.
+	numRequired := len(r.ringDesc.Ingesters)
+	if numRequired < r.cfg.ReplicationFactor {
+		numRequired = r.cfg.ReplicationFactor
+	}
+	numRequired -= r.cfg.ReplicationFactor / 2
```

Comment:
```suggestion
	maxUnavailable :=  r.cfg.ReplicationFactor / 2
	numRequired -= maxUnavailable
```

It took me a second to understand what `r.cfg.ReplicationFactor / 2` represented. This might make clearer to someone who hasn't looked at this before